ethcoder/typed_data_test.go

```diff
@@ -0,0 +1,102 @@
+package ethcoder
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/arcadeum/ethkit/ethwallet"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTypedDataTypes(t *testing.T) {
```

Comment:
Should probably add tests such that each basic type is being tested, e.g. the `uint` family, `bytes32`, `bytes`, etc.